src/effects/effectchainslot.cpp

```diff
@@ -429,15 +685,14 @@ void EffectChainSlot::slotControlChainPrevPreset(double v) {
 }
 
 void EffectChainSlot::slotChannelStatusChanged(const QString& group) {
-    if (m_pEffectChain) {
-        ChannelInfo* pChannelInfo = m_channelInfoByName.value(group, NULL);
-        if (pChannelInfo != NULL && pChannelInfo->pEnabled != NULL) {
-            bool bEnable = pChannelInfo->pEnabled->toBool();
-            if (bEnable) {
-                m_pEffectChain->enableForInputChannel(pChannelInfo->handle_group);
-            } else {
-                m_pEffectChain->disableForInputChannel(pChannelInfo->handle_group);
-            }
+    qDebug() << "slotChannelStatusChanged";
+    ChannelInfo* pChannelInfo = m_channelInfoByName.value(group, NULL);
+    if (pChannelInfo != NULL && pChannelInfo->pEnabled != NULL) {
+        bool bEnable = pChannelInfo->pEnabled->toBool();
+        if (bEnable) {
+            enableForInputChannel(pChannelInfo->handle_group);
```

Comment:
Let's get rid of these functions and move their code into this one.

Reply:
Ping. Either move these into one big function here or move the code for `enableForInputChannel` and `disableForInputChannel` next to this function in the file.